src/keria/app/agenting.py

```diff
@@ -857,17 +858,40 @@ def loadEnds(app):
 class BootEnd:
     """ Resource class for creating datastore in cloud ahab """
 
-    def __init__(self, agency):
+    def __init__(self, agency: Agency, username: str | None = None, password: str | None = None):
         """ Provides endpoints for initializing and unlocking an agent
-
         Parameters:
             agency (Agency): Agency for managing agents
-
+            username (str): username for boot request
+            password (str): password for boot request
         """
-        self.authn = authing.Authenticater(agency=agency)
+        self.username = username
+        self.password = password
         self.agency = agency
 
-    def on_post(self, req, rep):
+    def authenticate(self, req: falcon.Request):
+        if self.username is None and self.password is None:
```

Comment:
Maybe we should have `or` instead of `and` in case it's half configured.

Reply:
With the current implementation, you will get a 401 if it is half configured. This check returns without error if _neither_ username or password is configured. If anyone of them is configured, it must match the basic auth values. But perhaps there is a way to make that more clear?

Reply:
Which is arguably OK, because that is how you configured it.

Reply:
Yeah, that's fair enough. Nevermind then